sgl-kernel/csrc/cpu/mamba/fla.cpp

```diff
@@ -0,0 +1,386 @@
+#include "common.h"
+#include "vec.h"
+
+namespace {
+
+inline float softplus(float x, double threshold = 20.0) {
+  if (x > threshold)
+    return x;
+  else if (x < -threshold)
+    return std::exp(x);
+  else
+    return std::log1p(std::exp(x));
+}
+
+inline at::vec::Vectorized<float> softplus(const at::vec::Vectorized<float>& x, double threshold = 20.0) {
+  at::vec::Vectorized<float> mask_hi = x > at::vec::Vectorized<float>(threshold);
+  at::vec::Vectorized<float> mask_lo = x < at::vec::Vectorized<float>(-threshold);
+
+  at::vec::Vectorized<float> expx = x.exp();
+  at::vec::Vectorized<float> log1pex = (expx + at::vec::Vectorized<float>(1.0f)).log();
+
+  return at::vec::Vectorized<float>::blendv(at::vec::Vectorized<float>::blendv(log1pex, expx, mask_lo), x, mask_hi);
+}
```

Comment:
we can use `exp_u20` or `fexp_u20` as the inputs are actually bfloat16 and expanded to be float32.

https://github.com/pytorch/pytorch/blob/35d2da32bd88f10eac038d21c4f753b2bc171e1b/aten/src/ATen/cpu/vec/vec512/vec512_float.h#L315